modin/pandas/test/dataframe/test_map_metadata.py

```diff
@@ -571,6 +571,15 @@ def test_astype_category_large():
     assert modin_result.dtypes.equals(pandas_result.dtypes)
 
 
+def test_infer_objects_single_partition():
+    data = {"a": ["s", 2, 3]}
+    modin_df = pd.DataFrame(data).iloc[1:]
+    pandas_df = pandas.DataFrame(data).iloc[1:]
+    modin_result = modin_df.infer_objects()
+    pandas_result = pandas_df.infer_objects()
+    assert modin_result.dtypes.equals(pandas_result.dtypes)
```

Comment:
I was going to suggest that we use `df_equals` to check this but then realized that we do not check `dtypes` there I think...oops...

Reply:
this should eventually become a feature: https://github.com/modin-project/modin/issues/3804

Reply:
Yes, we should fix that soon, I think we might find a bunch of bugs.